kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/DeleteAndCreateHelper.java

```diff
@@ -40,15 +50,40 @@ public T deleteAndCreate(T item) {
     if (Boolean.FALSE.equals(deleted)) {
       throw new KubernetesClientException("Failed to delete existing item:" + item.getMetadata().getName());
     }
-    return createTask.apply(item);
+    try {
+      return createTask.apply(item);
+    } catch (KubernetesClientException e) {
```

Comment:
I don't think this is related to my original change to support PropagationPolicy, but my new integration test caught this issue. The default grace period is 30s, so unless you explicitly specify 0 seconds (which we don't here) it's possible that the DELETE call returns a 202 instead of 200, meaning it hasn't actually been deleted yet.

BaseOperation/OperationSupport/etc assume that a non-error response means the item was deleted. This is sort of true, but not really since it may have actually been queued for asynchronous deletion (per the 202 response). I looked into whether I could pipe the response Status through as the return value of the various `delete()` functions, but that proved to be a significant change (and probably a breaking one, since right now we expect to return a simple Boolean).

The simpler fix here seems to be to add a waiting period as I did here. Alternatively we could continually retry createTask for a period of time, but this might be less impactful on API servers?